modules/geo-layers/src/tile-layer/tile-2d-header.js

```diff
@@ -33,38 +38,43 @@ export default class Tile2DHeader {
     return result;
   }
 
-  loadData(getTileData) {
+  async _loadData(getTileData, requestScheduler) {
     const {x, y, z, bbox} = this;
-    if (!getTileData) {
+
+    const requestToken = await requestScheduler.scheduleRequest(this, tile => {
+      return tile.isSelected;
+    });
+
+    if (!requestToken) {
+      this._isCancelled = true;
       return;
     }
 
+    this._isCancelled = false;
     let tileData;
+    let error;
     try {
-      tileData = getTileData({x, y, z, bbox});
+      tileData = await getTileData({x, y, z, bbox});
     } catch (err) {
+      error = err || true;
```

Comment:
I believe err is always an object.

Reply:
It was failing from `Promise.resolve(null)`

https://github.com/visgl/deck.gl/pull/4645#discussion_r440432340

Reply:
*`Promise.reject(null)`